Bottom: c03bd1ac907421a7f693fbeb2bb277dab6c9c707
Top:    ddc5c818e666a8da92851a698568d5a0e9d1b06a
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 11:44:02 +0800

Refresh of autofs-5.0.7-fix-memory-leak-in-get_dc_list.patch

---

diff --git a/modules/dclist.c b/modules/dclist.c
index bd4d5e6..af21ce0 100644
--- a/modules/dclist.c
+++ b/modules/dclist.c
@@ -536,6 +536,7 @@ struct dclist *get_dc_list(unsigned int logopt, const char *uri)
 		else
 			strcat(tmp, " ");
 
+		list = NULL;
 		for (i = 0; i < numdcs; i++) {
 			if (i > 0)
 				strcat(tmp, " ");
@@ -549,8 +550,7 @@ struct dclist *get_dc_list(unsigned int logopt, const char *uri)
 					error(logopt,
 					      "invalid port: %u", dcs[i].port);
 					free_srv_rrs(dcs, numdcs);
-					/* Ensure realloced list is freed */
-					list = tmp;
+					free(tmp);
 					goto out_error;
 				}
 				strcat(tmp, port);
